### ISSUE TITLE
[CS8629]: LINQ .Where calls are not considered

### ISSUE THREAD

**Issue by deeprobin** [dotnet/roslyn#57757]
**Version Used**: 
.NET 5.0 / but might in .NET 6.0 the same

**Steps to Reproduce**:

```cs
                IEnumerable<long?> longValues = /* ... */;
                _ = tweet.UserMentions.Where(id => id.HasValue)
                    .Select(id => id.Value) // Here does CS8629 occur
                    .Distinct()
                    .Select(id => DoSomethingOther(id));
```

**Expected Behavior**:
Don't show this warning

**Actual Behavior**:
It shows this warning
![grafik](https://user-images.githubusercontent.com/10373452/141697101-8e486c5a-248d-45de-a0e8-0c59fd7f0a7c.png)

**Comment by CyrusNajmabadi** [comment]
Dupe of https://github.com/dotnet/csharplang/issues/3951.  Thanks!

**Comment by deeprobin** [comment]
@CyrusNajmabadi Thank you 😄
